crates/bevy_ecs/src/bundle.rs

```diff
@@ -144,22 +148,28 @@ impl BundleInfo {
             let component_id = *self.component_ids.get_unchecked(bundle_component);
             let component_status = bundle_status.get_unchecked(bundle_component);
             match self.storage_types[bundle_component] {
-                StorageType::Table => {
-                    let column = table.get_column(component_id).unwrap();
-                    column.set_unchecked(table_row, component_ptr);
-                    let column_status = column.get_ticks_unchecked_mut(table_row);
-                    match component_status {
-                        ComponentStatus::Added => {
-                            *column_status = ComponentTicks::new(change_tick);
-                        }
-                        ComponentStatus::Mutated => {
-                            column_status.set_changed(change_tick);
-                        }
+                StorageType::Table => match (component_status, overwrite_existing) {
+                    (ComponentStatus::Added, _) => {
+                        let column = table.get_column(component_id).unwrap();
+                        column.set_unchecked(table_row, component_ptr);
+                        let column_status = column.get_ticks_unchecked_mut(table_row);
```

Comment:
This line and the two above it should be pulled out of the match statement to avoid duplication.